### PR TITLE
Use non-github link for StevenBlack/hosts

### PR DESCRIPTION
**Pull request by lakshayg**
**By submitting this pull request, I confirm the following:** 

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**
Prevent downloading StevenBlack/hosts when there are no updates to the list

**How does this PR accomplish the above?:**
The github link for StevenBlack/hosts does not provide a `Last-Modified` header,
which requires the file to be downloaded everytime gravity is updated even if
there's no change to the file since the last update. The new link, taken from
the StevenBlack/hosts repository provides that header

**What documentation changes (if any) are needed to support this PR?:**
None

### CODE REVIEW

**Comment by PromoFaux** [comment]
My concern with this is that the non-github link is just plain `http`, which I feel is a step backwards.

**Comment by lakshayg** [comment]
I agree, that is a valid concern https://github.com/StevenBlack/hosts/issues/878

**Comment by PromoFaux** [comment]
Ah, OK. I've shared a link with him in that issue thread that may help move things along...!

**Review by yyyopio — APPROVED** [review]

**Review by yyyopio — APPROVED** [review]

**Comment by DL6ER** [comment]
Unfortunately, still no movement to HTTPS. Also, I'll rather not merge this PR as the new adlist target has been mentioned as an "aging Windows server" the maintainer would rather like to see going silent forever.

It gives me the feeling that if the server fails for some reason (e.g., hardware failure), it will not return into operation - or at least not quickly. This would leave the majority of users with an forever outdated list. We can make a new PR once HTTPS is available upstream (we should all get notified by Github if he writes something in there).

**Pull request closed without merging**
